ScoutSuite/providers/azure/authentication_strategy.py

```diff
@@ -101,55 +64,32 @@ def authenticate(self,
         try:
 
             # Set logging level to error for libraries as otherwise generates a lot of warnings
-            logging.getLogger('adal-python').setLevel(logging.ERROR)
-            logging.getLogger('msrest').setLevel(logging.ERROR)
-            logging.getLogger('msrestazure.azure_active_directory').setLevel(logging.ERROR)
-            logging.getLogger('urllib3').setLevel(logging.ERROR)
-            logging.getLogger('cli.azure.cli.core').setLevel(logging.ERROR)
+            logging.getLogger('azure.identity').setLevel(logging.ERROR)
+            logging.getLogger('azure.core.pipeline').setLevel(logging.ERROR)
 
             context = None
 
             if cli:
-                arm_credentials, subscription_id, tenant_id = \
-                    get_azure_cli_credentials(with_tenant=True)
-                aad_graph_credentials, placeholder_1, placeholder_2 = \
-                    get_azure_cli_credentials(with_tenant=True, resource='https://graph.windows.net')
+                identity_credentials = AzureCliCredential()
 
             elif user_account:
 
-                if not (username and password):
+                if not (username and password and tenant_id):
                     if not programmatic_execution:
+                        tenant_id = tenant_id if tenant_id else input("Tenant ID: ")
                         username = username if username else input("Username: ")
                         password = password if password else getpass("Password: ")
                     else:
-                        raise AuthenticationException('Username and/or password not set')
+                        raise AuthenticationException('Username, Tenant ID and/or password not set')
```

Comment:
Did you check that https://github.com/nccgroup/ScoutSuite/blob/e3850327067b729baf3a3b4cac4b604d571439ae/ScoutSuite/core/cli_parser.py#L394 reflects the new required parameters?

Reply:
Previously, for user-account, we were asking the tenant id after inputting the command. Now the workflow will be identical to user-account-browser and the user has to put --tenant parameter in the cli

Reply:
So shouldn't those checks be added to the CLI parsing?

Reply:
Yes, now for user-account, we have added verification for the --tenant parameter as well. This exception will be switched back to 'Username and/or password not set'